### ISSUE TITLE
Taxes: the way to go.

### ISSUE THREAD

**Issue by carlosescri** [siwapp/siwapp#75]
In settings, allow the user to create _Tax Groups_ that group multiple taxes under the same label.

In the forms, use a one-choice `<select>` tag with two option groups, one for single taxes and one for tax groups:

![image](https://cloud.githubusercontent.com/assets/482075/13494396/354be532-e144-11e5-97cc-1403a2c7c105.png)

**Comment by serguitus** [comment]
Adding Tax Groups in addition to Taxes makes the relation more complex. We would need to create a generalization model (apart from tax-group model) to make the relation with items so they can reference a tax or a tax group. I think that it would be easier to relate always tax groups. then we could change the tax creation page with a Tax-Group creation page and there you could add taxes to each group. Of course, you could add 1-item groups and the select would just have tax-gropus.
I'm already working on this idea but I would like to hear @carlosescri thoughts (or any other's) on this
The other option that comes to my mind to apply the original proposal is doing some processing before saving invoices to translate tax-groups into taxes and then save
I'm all ears

**Comment by carlosescri** [comment]
If you translate tax groups into single taxes you will loose the UI enhancements (having a single select).

We are currently using a common model to hold invoices and recurring invoices, and we could do the same with taxes.

One tax would be linked to other taxes and that would make it act as a tax group. Of course you can't link a tax with itself or with another groups but that is validation stuff. 

When you create a tax that links IVA21% and IRPF-15% then its percentage becomes the sum of the percentages of its components. In this case will be 6%.

Take into account that I don't have my head in Ruby right now :-)

**Comment by ecoslado** [comment]
I was doing a scaffolding for TaxGroup and we've noticed that is too much effort and resources for a simple select, so I just changed the way the taxes are shown. I put them in checkboxes, where default are checked. Maybe this could be replaced by a popup. This behaviour is enough for users who usually apply the same taxes. TaxGroup would be useful for those users who apply different tax sets depending on the invoice or the line.
